### PR TITLE
Fixed errors when viewing profile as anonymous user

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
- fixes https://github.com/mitodl/micromasters/issues/1837

#### How should this be manually tested?
- Copy learners page link i.e http://192.168.99.100:8079/learner/amirq
- open it in incognito or any process we MM session is not in progress.

@pdpinch @noisecapella 
#### Screenshots (if appropriate)
screen shot 2016-12-22 at 5 15 55 pm

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks good, no console errors!

File: static/js/containers/UserPage_test.js [hunk 1]
```diff
@@ -1042,30 +1042,37 @@ describe("UserPage", function() {
...
       helper.profileGetStub.
         withArgs(USER_PROFILE_RESPONSE.username).
         returns(Promise.resolve(USER_PROFILE_RESPONSE));
+      user = SETTINGS.user;  
       SETTINGS.user = null;
     });
 
     afterEach(() => {
       helper.cleanup();
+      SETTINGS.user = user;
```
Comment (aliceriot) [inline static/js/containers/UserPage_test.js]: why is this necessary?

Comment (noisecapella) [inline static/js/containers/UserPage_test.js]: It shouldn't be, all of `SETTINGS` is reset after each test in `global_init.js`

Comment (aliceriot) [inline static/js/containers/UserPage_test.js]: Yeah that's what I thought

**Comment by amir-qayyum-khan** [comment]
Fixed @aliceriot
